### PR TITLE
Implement GraphQL Dashboard API

### PR DESCRIPTION
**Pull request by jthurman42**
*Note:* Dashboard API via NerdGraph is in closed beta, this PR will not work for everyone at this time.

Refactor `newrelic_dashboard` to use the new GraphQL API, enabling all new dashboard features (pages/multi-query/etc)

### CODE REVIEW

File: newrelic/provider.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ func Provider() terraform.ResourceProvider {
 			"newrelic_infra_alert_condition":                    resourceNewRelicInfraAlertCondition(),
 			"newrelic_insights_event":                           resourceNewRelicInsightsEvent(),
 			"newrelic_nrql_alert_condition":                     resourceNewRelicNrqlAlertCondition(),
+			"newrelic_one_dashboard":                            resourceNewRelicOneDashboard(),
```
Comment (sanderblue) [inline newrelic/provider.go:147]: Did we inquire marketing and/or other sales folks on this name?  


Disclaimer: I'm on board with this resource name ✔️

File: newrelic/resource_newrelic_one_dashboard.go [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The dashboard's description.",
+			},
+			"permissions": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "public_read_only",
+				ValidateFunc: validation.StringInSlice([]string{"private", "public_read_only", "public_read_write"}, false),
```
Comment (sanderblue) [inline newrelic/resource_newrelic_one_dashboard.go:51]: Not a blocker, but we've usually tried to make things case-insensitive for constants like these (per JT suggestions 😉 ), but I also see the merit with maintaining a slightly stricter API.

File: newrelic/provider.go (hunk 1 above)
Comment (jthurman42) [reply newrelic/provider.go:147]: This is the name that was landed at after talking with some marketing folks.

File: newrelic/resource_newrelic_one_dashboard.go (hunk 2 above)
Comment (jthurman42) [reply newrelic/resource_newrelic_one_dashboard.go:51]: They must be UPPERCASE for the API, and are upcase/tolower in the expand/flatten funcs, so updating the validator to not care is fine with me.

**Review by sanderblue — APPROVED** [review]
🐑 it

**Pull request merged**